### DISCUSSION TITLE
Notification ScheduleDate

### DISCUSSION THREAD

**Discussion by sandeepc24** [shinyorg/shiny#985] [Q&A] [answered]
In `Notification` is `ScheduleDate` a UTC value or a local timezone value?

**Comment by sandeepc24** [comment]
I am seeing strange behaviour when I set `ScheduleDate` in that the notification is not shown all the time , if I don't set this field then the notification is always shown correctly.

I have set `ScheduleDate` using DateTimeOffset.UtcNow.AddSeconds(2).

**Reply by aritchie** [reply]
Because there is no job waiting to pick up the schedule.  You are sending it immediately

**Comment by sandeepc24** [comment]
Another thing I noticed is that the notification is shown the next time the app is started.

**Reply by aritchie** [reply]
Assuming this is on android, yes - you are killing all of the services running which means the job has to pick up the slack.  The job is run everytime the app starts.  Please review the code to see how this works.

**Comment by sandeepc24** [comment]
It looks like there is a delay between when the notification is scheduled for to when it is actually shown.

**Reply by aritchie** [reply]
There will be.  On Android, the job has to run which has a delay of 15mins.  I am switching to exact timers in 2.5, but that also has a delay

**Comment by aritchie** [comment, marked as answer]
@sandeepc24 please understand that this is not a product.  Please make sure to review the code instead of sending a wad of questions.

**Comment by sandeepc24** [comment]
Apologies for bombarding with questions.
